test/elm/convert/test.coffee

```diff
@@ -142,34 +142,49 @@ describe 'FromDateTime', ->
   @beforeEach ->
     setup @, data
 
-  it "should convert @2015-01-02 to '2015-01-02'", ->
-    @dateStr.exec(@ctx).should.equal "2015-01-02"
+  it "should convert @2015-01-02T12:01:02.321-06:00 to '2015-01-02T12:01:02.321-06:00'", ->
+    @dateTimeToStr.exec(@ctx).should.equal "2015-01-02T12:01:02.321-06:00"
 
-  it "should convert @2015-01-02 to @2015-01-02", ->
-    date = @dateDate.exec(@ctx)
+  it "should convert @2015-01-02T12:01:02.321-06:00 to Date", ->
+    date = @dateTimeToDate.exec(@ctx)
+    date.isDate.should.be.true
     date.year.should.equal 2015
     date.month.should.equal 1
     date.day.should.equal 2
+    should.not.exist(date[field]) for field in [ 'hour', 'minute', 'second', 'millisecond', 'timezoneOffset' ]
+
+  it "should convert @2015-01-02T12:01:02.321-06:00 to DateTime", ->
+    dateTime = @dateTimeToDateTime.exec(@ctx)
+    dateTime.isDateTime.should.be.true
+    dateTime.year.should.equal 2015
+    dateTime.month.should.equal 1
+    dateTime.day.should.equal 2
+    dateTime.hour.should.equal 12
+    dateTime.minute.should.equal 1
+    dateTime.second.should.equal 2
+    dateTime.millisecond.should.equal 321
+    dateTime.timezoneOffset.should.equal -6
 
 describe 'FromDate', ->
   @beforeEach ->
     setup @, data
 
-  it "should convert ToDate('2015-01-01') to DateTime", ->
+  it "should convert @2015-01-01 to DateTime", ->
     dateTime = @dateDateTime.exec(@ctx)
     dateTime.year.should.equal 2015
     dateTime.month.should.equal 1
     dateTime.day.should.equal 1
+    dateTime.timezoneOffset.should.equal @ctx.getTimezoneOffset()
```

Comment:
This test should also check that all the time components (hour, minute, second, millisecond) are `0` (as specified in [Section 3.17](http://cql.hl7.org/09-b-cqlreference.html#todatetime) of the spec).

Looking at our current implementation ([here](https://github.com/cqframework/cql-execution/blob/master/src/elm/type.coffee#L63-L64)  and [here](https://github.com/cqframework/cql-execution/blob/master/src/datatypes/datetime.coffee#L503-L504)), I think that test would actually fail.  I know that's not code you touched in this PR, but you touched the test. ;-)

I actually should have caught this one before.

Reply:
Good point. Fixed.